openedx/openedx-platform
# create course_roles on django admin dashboard

## Description

This code adds a section to the LMS django dashboard for assigning course_roles on a course, organization, or instance level. This code includes a db migration to allow blank values for user roles. No roles have been created yet, so there are no significant data considerations with this change.

<img width="1414" alt="Screenshot 2023-10-27 at 9 21 20 AM" src="https://github.com/openedx/edx-platform/assets/10408711/a34343e6-52c9-44d2-8d6f-384355536c62">


## Testing instructions

The change can be tested by running the code locally and visiting http://localhost:18000/admin/course_roles/courserolesuserrole/add/. It is necessary to create a role in order to test this code and then the following scenarios can be tested.
- create role with email, course, org, role (course level role)
- create role with email, org, role (org level role)
- create role with email, role (instance level role)
- attempt to create a role with email, course, role (should show error message)
- attempt to re-create a role that has already been created (should show error message)
- attempt to create a role with email, course, org, role, but choose the wrong org, one that doesn't match the org on the class record (should show error message)
